Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.DataGridViewHeaderCell.cs

```diff
@@ -0,0 +1,249 @@
+        public override DataGridViewElementStates GetInheritedState(int rowIndex)
+        { 
+            Contract.Requires(this.OwningRow == null ||
+                              !((this.DataGridView == null && rowIndex != -1 ||
+                                 this.DataGridView != null && (rowIndex < 0 || rowIndex >= this.DataGridView.Rows.Count))));
+            Contract.Requires(this.OwningColumn == null || this.DataGridView == null || rowIndex == -1);
+
+            return default(DataGridViewElementStates);
+        }
+
+        /// <summary>
+        /// Gets the size of the cell.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// A <see cref="T:System.Drawing.Size"/> that represents the size of the header cell.
+        /// </returns>
+        /// <param name="rowIndex">The row index of the header cell.</param><exception cref="T:System.ArgumentOutOfRangeException">The value of the <see cref="P:System.Windows.Forms.DataGridViewElement.DataGridView"/> property for this cell is null and <paramref name="rowIndex"/> does not equal -1. -or-The value of the <see cref="P:System.Windows.Forms.DataGridViewCell.OwningColumn"/> property for this cell is not null and <paramref name="rowIndex"/> does not equal -1. -or-The value of the <see cref="P:System.Windows.Forms.DataGridViewCell.OwningRow"/> property for this cell is not null and <paramref name="rowIndex"/> is less than zero or greater than or equal to the number of rows in the control.-or-The values of the <see cref="P:System.Windows.Forms.DataGridViewCell.OwningColumn"/> and <see cref="P:System.Windows.Forms.DataGridViewCell.OwningRow"/> properties of this cell are both null and <paramref name="rowIndex"/> does not equal -1.</exception><exception cref="T:System.ArgumentException">The value of the <see cref="P:System.Windows.Forms.DataGridViewCell.OwningRow"/> property for this cell is not null and <paramref name="rowIndex"/> indicates a row other than the <see cref="P:System.Windows.Forms.DataGridViewCell.OwningRow"/>.</exception>
+        protected override Size GetSize(int rowIndex)
+        {
+            Contract.Requires(this.OwningRow == null ||
+                              !((this.DataGridView == null && rowIndex != -1 ||
```

Comment:
Because this is requires, this check should be inverted. You inverted only the first check (`OwningRow == null`). I don't think that this code is correct!